### PR TITLE
fix(decode): reserve >= dflash_block_size outputs for DFlash draft contexts (#108)

### PR DESCRIPTION
**Pull request by marksverdhei**
**Draft** — the fix is complete and the diagnosis is high-confidence, but it must be smoke-tested against the cluster DFlash model before un-drafting (no local repro — `gemma-4-31b-dflash-Q6_K` is 62 G, doesn't fit here).

## Root cause (full trace in #108)
The DFlash draft decodes the entire diffusion block in one `llama_decode` with every token `logits=true` (`common/speculative.cpp:1325-1331`), so `n_outputs_all == dflash_block_size` (default 16). But the server sizes the draft context's output buffer to `n_parallel` (`tools/server/server-context.cpp:933`, unconditional), so `output_reserve(16)` trips `GGML_ASSERT(n_outputs_max <= cparams.n_outputs_max)` (`llama-context.cpp:2407`) on the **first** `draft()` → the "loads clean, child dies on first decode" crash. MTP survives the same line because it outputs `n_parallel` per step; DFlash needs `>= block_size`.

## Fix
Clamp `cparams.n_outputs_max` up to `dflash_block_size` in the `llama_context` ctor for DFlash-arch models:
```cpp
if (model.arch == LLM_ARCH_DFLASH) {
    cparams.n_outputs_max = std::max(cparams.n_outputs_max, model.hparams.dflash_block_size);
}
```
In-library guard → defends every caller, not just the server path.

## Validation
- ✅ Compiles clean (`llama-server` links; pre-existing clang-tidy warnings only, none on the change).
- ✅ Non-DFlash server load + `/completion` works (the ctor change is a no-op for non-DFlash — guarded on `model.arch`).
- ⏳ **Pending:** cluster smoke against `gemma-4-31b-dflash-Q6_K` — load + first `draft()` should now succeed instead of `GGML_ASSERT`. Un-draft once green.

Filed per `/home/me/IDLE.md` (draft-PR blocked-on-verification items rather than hold).

### CODE REVIEW

**Comment by marksverdhei** [comment]
Cross-confirmation from a bug-hunt of the adjacent DFlash multi-seq path: DFlash **forces `--parallel 1`** — the server refuses to start otherwise (`server-context.cpp:1022-1027`, because `dflash.target_features` isn't keyed per seq_id). So the draft context's `n_outputs_max` (= `n_parallel`, `server-context.cpp:933`) is **always exactly 1**, and `output_reserve(16)` asserting `16 <= 1` makes the first-decode crash **deterministic**, not just 'possible when n_parallel < block_size'. That removes any doubt about repro conditions — DFlash + this server path crashes 100% of the time on first `draft()`, which matches the #108 report. The clamp in this PR (`n_outputs_max >= dflash_block_size`) is necessary and sufficient.

**Comment by marksverdhei** [comment]
Superseded by #121 ("clamp n_outputs_max for DFlash models"), which is merged and lands the identical fix — `src/llama-context.cpp` now has the same `cparams.n_outputs_max = std::max(cparams.n_outputs_max, model.hparams.dflash_block_size)` for `LLM_ARCH_DFLASH`. Issue #108 is closed. The only remaining delta in this branch is an explanatory comment; not worth a separate PR. Closing as redundant.
